ilastik_scripts/ilastik_startup.py

```diff
@@ -138,9 +143,59 @@ def find_library(name):
     ctypes.util.find_library = find_library
 
 
+def fix_ssl() -> None:
+    """
+    Paths to cert.pem and the CA certificates dir are hard-coded into openssl during the build process
+    of our installer packages.
+    It has some special logic for Windows, and there the paths don't seem to be an issue.
+    On Linux and macOS, the build machine's path is written (/opt/conda/envs/ilastik-release/ssl/),
+    which does not exist on the end user's machine.
+    Being unable to find the cert files means aiohttp cannot connect via https. Currently only
+    zarr.storage.FSStore depends on this.
+    We override the broken paths using the env vars that openssl listens to. Better would be if we can
+    find a way to write the correct paths during build.
+    """
+    if platform.system() == "Windows":
+        return
+
+    ssl_paths = ssl.get_default_verify_paths()
+    if os.path.isfile(ssl_paths.openssl_cafile):
+        # All good then I guess
+        return
+
+    def check_parent_for_dir_recursive(path: pathlib.Path, dir_name: str) -> Optional[pathlib.Path]:
+        if path.is_dir() and any(p.name == dir_name for p in path.iterdir()):
+            return path
+        if path.parent == path:
+            raise RecursionError(f"Could not find {dir_name} directory")
+        return check_parent_for_dir_recursive(path.parent, dir_name)
+
+    ssl_parent = None
+    try:
+        ssl_parent = check_parent_for_dir_recursive(pathlib.Path(__file__).parent, "ssl")
+    except RecursionError:
+        try:
+            # Try from PWD and OLDPWD env variables
+            # on macOS: os.environ["PWD"] = {app_root}/Contents/MacOS; os.environ["OLDPWD"] = {app_root}/Contents
+            # on Linux: os.environ["PWD"] = {tar_root}; os.environ["OLDPWD"] = {tar_root}/bin
+            app_root_or_subdir = pathlib.Path(os.environ.get("PWD", "") or os.environ.get("OLDPWD", ""))
```

Comment:
why not use `pathlib.Path.cwd()` ?

Also, do we have much to gain from looking from `CWD`? Will we ever find anything there? Shouldn't we look in the places where the certs could actually be, like `/etc/pki/tls/cert.pem` or something like it?

Reply:
Honestly didn't think of `cwd`, I was just collecting all env variables I have on Mac and Linux and seeing where I can get a path to the package root 😅 Will have to check what `cwd` actually returns on Linux and Mac when running from the package.

Instead of guessing where on the user's system their system certs might be, I was going for the certs that the conda env comes with. In the Linux tarball, they're in `$tar_root/ssl` and on Mac they're in `$app_root/Contents/ssl` (iirc, will have to check again)

Reply:
Decided to remove the cwd attempt altogether because on Linux cwd/PWD could literally be anything (wherever the user called run_ilastik.sh from); on Mac it would be quite likely to be the app directory, but I would end up hard-coding all sorts of relative path shenanigans.
I'm verifying now that `__file__` actually is inside `ilastik_release` on Mac, i.e. that we will actually discover `ilastik_release/ssl` by traversing up the tree